source/fe/fe_values.cc

```diff
@@ -471,7 +472,7 @@ namespace FEValuesViews
            shape_function<dofs_per_cell; ++shape_function)
         if (shape_function_data[shape_function].is_nonzero_shape_function_component)
           {
-            const Number value = dof_values(shape_function);
+            const Number value = dof_values[shape_function];
```

Comment:
For scalars, temps are actually simpler for the compiler. Are you concerned about things such as AD types?